content/post/hercules.v4.md

```diff
@@ -0,0 +1,295 @@
+---
+author: vadim
+date: 2018-08-29
+title: "MLonGit: Hercules v4 released"
+image: /post/hercules.v4/intro.png
+description: "src-d/hercules is an open source project started in late 2016 with the goal to speed up collecting line burndown statistics from Git repositories. It has transformed into a general purpose Git repository mining framework with several cool use cases: ownership through time, file and people embeddings, structural hotness and even comment sentiment estimation. The post presents the latest 'v4' release of Hercules and gives some insights about how Git works."
+categories: ["MLonCode", "technical"]
+---
+
+This post follows up [the first one from January 2017](../hercules). Please read it
+if you wonder why Hercules was named Hercules, what is a line burndown plot and how
+everything started.
+
+# v4 features
+
+The main features of the recent release:
+
+1. Hercules is now aware of forks and merges and thus visits all the commits in the history, not just the main linear sequence as before.
+2. External plugins.
+3. Merging results from multiple repositories together.
+4. More "batteries" included: ownership, files, people, structure embeddings, comment sentiment.
+
+### Forks and merges
+
+The difference is the best to show rather than explain:
+
+{{% caption src="/post/hercules.v4/forks_merges.gif" %}}
+Prior to v4, Hercules did not process the full commit history and worked with it as a linear sequence.
+{{% /caption %}}
+
+This feature yields an improvement of the analysis accuracy. However, nothing
+is free unfortunately: processing all the commits can considerably slow down the analysis.
```

Comment:
a comma may be needed with the interrupted `unfortunately` - `nothing is free, unfortunately`